packages/components/src/components/stepper-horizontal/stepper-horizontal-item/stepper-horizontal-item-styles.ts

```diff
@@ -74,96 +71,99 @@ export const getComponentCss = (state: StepperState, disabled: boolean, theme: T
           ...hostHiddenStyles,
           '&(:not(:last-of-type))': {
-            marginRight: '1em',
+            margin: '0 16px 0 0',
           },
         }),
       },
       button: {
+        display: 'grid',
         position: 'relative',
+        gridTemplateColumns: `${fontLineHeight} minmax(0, auto)`,
+        gap: '3px',
         color: isDisabled ? disabledColor : primaryColor,
-        transition: getTransition('color'),
-        margin: 0,
-        padding: `0 0 0 calc(${fontLineHeight} + ${pxToRemWithUnit(4)})`, // icon height + 4px
+        padding: '4px 10px 4px 6px',
         background: 0,
         border: 0,
+        outline: 0,
         ...textSmallStyle,
         fontSize: 'inherit',
-        whiteSpace: 'nowrap',
-        ...getFocusJssStyle(),
-        ...(isStateCurrentOrUndefined
-          ? // counter circle icon via css
-            {
-              cursor: isDisabled ? 'not-allowed' : 'auto',
-              // TODO: combine &::before and &::after element
-              '&::before': {
-                // circle of counter element
-                content: '""',
-                position: 'absolute',
-                top: '50%',
-                transform: 'translateY(-50%)',
-                left: counterCirclePosition,
-                height: counterCircleSize,
-                width: counterCircleSize,
-                borderRadius: '50%',
-                ...(isDisabled
-                  ? {
-                      boxSizing: 'border-box',
-                      border: `1px solid ${disabledColor}`,
-                    }
-                  : {
-                      background: primaryColor,
-                    }),
-              },
-              '&::after': {
-                // value of counter element
-                content: '""',
-                position: 'absolute',
-                top: '50%',
-                transform: 'translateY(-50%)',
-                left: counterValuePosition,
-                width: counterValueSize,
-                height: counterValueSize,
-                background: `${getSvg(
-                  isDisabled ? disabledColor : invertedBaseColor
-                )} 0 50% / ${spriteWidth} ${spriteHeight} no-repeat`,
-              },
-            }
-          : // other icons via icon component
-          isDisabled
-          ? {
-              cursor: 'not-allowed',
-              textDecoration: 'none',
-            }
-          : {
-              cursor: 'pointer',
-              textDecoration: 'underline',
-              ...hoverMediaQuery({
-                ...getHoverJssStyle(),
-                '&:hover .icon': {
-                  color: hoverColor,
-                },
+        width: 'max-content',
+        cursor: isDisabled ? 'not-allowed' : 'pointer',
+        // mergeDeep needed because of hoverMediaQuery in certain modes not wrapping keys and therefore overriding "&::before" key
+        ...mergeDeep(
+          {
+            '&::before': {
```

Comment:
An alternative to the background image on the button would be to use this `::before` pseudo element for the background svg and then use a natural flex flow from left to right. This could then be shared with the icon without any absolute positioning.
The focus could then be moved to the `::after` pseudo element.